flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/JobValidationError.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.dispatcher;
+
+import org.apache.flink.annotation.Internal;
+
+/** Captures errors during the execution of the user jar's main method. */
+@Internal
+public interface JobValidationError {
+    String getErrorMessage();
+}
```

Comment:
Why does it make sense to use an interface here?

Reply:
I thought in the future it might become handy if we want to add more exceptions besides the `ConfigurationNotAllowedError`. I can also remove it. WDYT?